cmd/panda_test_executor/panda_executor_test.go

```diff
@@ -0,0 +1,456 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	controller "github.com/panda-re/panda_studio/internal/panda_controller"
+)
+
+// Each enum represents the state panda was in that caused the exception
+// Enum should match that in /panda_agent/agent.py
+const (
+	RUNNING       = 0
+	NOT_RUNNING   = 1
+	RECORDING     = 2
+	NOT_RECORDING = 3
+	REPLAYING     = 4
+	NOT_REPLAYING = 5
+)
+
+var error_to_string [6]string = [6]string{"RUNNING", "NOT_RUNNING", "RECORDING", "NOT_RECORDING", "REPLAYING", "NOT_REPLAYING"}
+
+// Extracts the error number from an err from agent
+func getError(err error) int {
+	// Find the numbers in the error message
+	re := regexp.MustCompile("[0-9]+")
+	// Return the first one as an integer
+	nums := re.FindAllString(err.Error(), -1)
+	num, _ := strconv.Atoi(nums[0])
+	return num
+}
+
+var num_passed int = 0
+var num_tests int = 0
+
+// Runs a test for the agent, recording and replay
+// Prints the number of tests and success rate
+func TestMain(t *testing.T) {
```

Comment:
Similar to the enum comment, we should include an example of how to set up a test somewhere or give a break down on what to do so that others can add tests. Most specifically the t.cleanup() function is likely hard to understand for some programmers

Reply:
Added a readme with that info